Adding a 3D Printabale model of the DotNet Bot

CliffAgius opened the pull request:
Printable STL File of the DotNet Bot that can be used my all 3D Printers, also included the 3mf file which is the new more detailed project file for 3D printers that most new slicers can use.

Image included with the readme to show what it looks like when printed.

Have fun printing.

drewnoakes commented:
Awesome! Just loaded the .3mf file in Cura and it's showing model errors in the belt buckle. Is this a problem in practice?

![image](https://user-images.githubusercontent.com/350947/95507199-cbc49a00-09fc-11eb-879c-77f35958aa2a.png)

The polka-dot area is being highlighted. Perhaps the normals are inverted there?

I've only seen this once before and was still able to print the object, so perhaps it's not an issue. Thanks for sharing the model.

CliffAgius commented:
@drewnoakes yes the belt is not printing hence it's missing in my picture we know what is wrong just not had time to fix it yet and to be honest most printers can't print that accurate hence doing the PR while we fix the issue.  If you slice it will print but the belt buckle won't be attached and will fall off so we need to embed it a little more into the body of the Bot.  I hope to get some time in the next few days and will update the PR or create a new one it's just I tweeted about the print and got lots of requests for it so thought it should live on the Foundation GitHub rather than mine as I'm not the owner of the DotNet Bot image/artwork etc.

Let me know if it's ok or not.

drewnoakes commented:
I think putting the model up here is a great idea and I'm definitely going to print one, so thanks very much for driving this. From my perspective this is good to merge as-is.

If you are going to make further changes, consider also filling the hollow void. My slicer wants to put support structures in there (both above and below the horizontal shelf), which I won't be able to remove and which may come loose and rattle inside.

![image](https://user-images.githubusercontent.com/350947/95508797-42629700-09ff-11eb-849b-704cea225dc0.png)

CliffAgius commented:
@drewnoakes we have updated the Model now so that it has the belt we found the missing vertices and corrected them in Blender and we also made the belt a little bigger so that it prints better.  We have one printing on our Prusa printer in Multi-Material mode but it slices a lot better now.

Let me know if you need anything else to get this PR accepted.

Thanks

drewnoakes commented on dotnet-bot-3DPrint/README.md:
```diff
@@ -2,17 +2,21 @@
-This folder contains the 3D STL files that can be used to 3D print a nice little DotNet Bot, here is one printed earlier.
+![DotNet Bot Model](DotNetBot-3DModel.jpg)
+
+This repo contains the 3D STL files that can be used to 3D print a nice little DotNet Bot, here is one printed earlier.
```
```suggestion
This folder contains the 3D STL files that can be used to 3D print a nice little DotNet Bot, here is one printed earlier.

```

drewnoakes reviewed (approved):
Tested the new model out in Cura and it looks much better, thanks!

This PR should be squash merged (or interactive rebased), as it has revisions of a large (27MB) binary file that are not needed in the history and will bloat the database.

CliffAgius commented:
@drewnoakes yep your right sorry didn't think about that I but here you go and thanks for the suggestion on the ReadMe change that's been added as well.

BethMassi commented:
Thank you so much for submitting this, @CliffAgius! @annasoracco and @kawilkin are also working on a 3D dotnet-bot model for some digital Swag we're building out for the .NET Conf event (see: https://github.com/dotnet-presentations/dotNETConf/tree/master/2020/MainEvent). Maybe you all can collaborate?

CliffAgius commented:
@BethMassi we are working on giving the DotNet Bot a nice little surf board at the moment. I'm giving a Xamarin/MAUI talk at the conference and wanted it printed and in the background of I get approval. Happy to work with the team.

theuserbl commented:
Nice. But for me it looks not so much as the original bot.

I have tried out a little bit with a more original bot in 3D. But I am not a designer. And I have no idea how to create the finbgers and feets. And my complete bot is in the implementation suboptimal. But the idea is good.
Btw. my bot is also under the CC0 1.0 public domain license.

[source.zip](https://github.com/dotnet-foundation/swag/files/8935924/source.zip)
<img width="377" alt="mybot1" src="https://user-images.githubusercontent.com/8337032/174490950-08c94a60-f84a-4f85-8763-3718d7d7357b.png">
<img width="647" alt="mybot2" src="https://user-images.githubusercontent.com/8337032/174490951-072094dd-40bf-4398-b8e9-45886e3431fe.png">
<img width="572" alt="mybot3" src="https://user-images.githubusercontent.com/8337032/174490948-b0330ec6-db1c-4df3-881d-236df60f1d81.png">

theuserbl commented:
Only for completeness. I have now seen, that on Thingiverse is also a .net Bot. The head is round like a ball, but all other parts have limited mashes.
And as the Thingivwerse side says, it is under the CC BY 4.0 license.

https://www.thingiverse.com/thing:305857
![bc1225c3ac2c8cd4a12dcf07213d42ef_preview_featured](https://user-images.githubusercontent.com/8337032/174494065-fb2cdbd0-9296-4192-8c1a-00bbd60450ae.jpg)
Looks like created with clay.

theuserbl commented:
@CliffAgius In your submitted Readme stands

> This model was designed by Clifford Agius and Modelled in Blender by Macauley Agius as a fun project and we hope you enjoy printing lots of them.

So it was created in Blender. Do you have still the Blender source of it?
The STL files haven't any coloring.
And the 3MF files are completely not supported by Blender. There are also no addons for it.

theuserbl commented:
Here the personal repo of the thread opener of .net bot.
It includes an image, where he have printed its .net bot.

theuserbl commented:
There is a presentation of making a .NET Bot.
https://www.mlclm.me/projects/7236511#3
It seems, that it was created by a Microsoft Employee.

The source of this .net bot can be found at
https://github.com/dotnet-presentations/dotNETConf/tree/master/2021/MainEvent/DigitalSWAG/3D/dotnet-bot
![dotnet-bot-3D_defaultpose](https://user-images.githubusercontent.com/8337032/174681235-5c6fb88f-d0ca-4fc0-b87f-d610ba22e665.png)

theuserbl commented:
@drewnoakes When do you plan to integrate some 3D versions of the .NET Bot in this repo?
Seems, that there existing some 3D models.

theuserbl commented:
Ok, then I make this here to a central point for .net Bot images.

At
https://dotnetcore.show/
is a modified version of the first .net Bot.
![logo2](https://user-images.githubusercontent.com/8337032/175102250-bcb8eeca-16e6-43b5-b73d-87cb243f07f1.svg)

There are a lot os 2D .net Bots
https://swimburger.net/media/dikleklg/dotnet-bot_builder.svg
https://devblogs.microsoft.com/dotnet/wp-content/uploads/sites/10/2021/09/dotnet-bot_jetpack-faceing-right.png
https://swimburger.net/media/ifzbm55o/dotnet-framework-square.png
https://upload.wikimedia.org/wikipedia/commons/thumb/4/43/Dotnet-bot.svg/848px-Dotnet-bot.svg.png

They are possibly based on and come are created with the .net bot creator
https://mod-dotnet-bot.net/
https://github.com/dotnet-foundation/mod-dotnet-bot
![my-dotnet-bot-mod-1](https://user-images.githubusercontent.com/8337032/175103102-a2d9496e-f484-4f32-a9a5-5f95a5bfcc3c.png)
Btw. looks the last posted 3D .net Bot like the 2D Bots of the .net Bot creator.


And the first .net Bot have a green twin.
![pixar](https://user-images.githubusercontent.com/8337032/175103760-b86f15c8-ec19-4fe7-abb9-7465f5c5024e.jpg)
It seems, that at first a company have created the green bot. And the company have created a lot of more robots, which looks similar like that. And Microsoft or the .NET Foundation have buyed a license of it, changed the color to violet and published it as public domain.

theuserbl commented:
A lot of 2D .net Bot 2.0 graphics, which are not created with the creator, are at
https://github.com/dotnet/brand/tree/main/dotnet-bot-illustrations
